contentcuration/contentcuration/models.py

```diff
@@ -970,6 +988,118 @@ def get_nodes_with_title(cls, title, limit_to_children_of=None):
         else:
             return cls.objects.filter(title=title)
 
+    def get_details(self):
+        """
+        Returns information about the node and its children, including total size, languages, files, etc.
+
+        :return: A dictionary with detailed statistics and information about the node.
+        """
+        descendants = self.get_descendants().prefetch_related('children', 'files', 'tags') \
+            .select_related('license', 'language')
+        channel = self.get_channel()
+
+        # If channel is a sushi chef channel, use date created for faster query
+        # Otherwise, find the last time anything was updated in the channel
+        last_update = channel.main_tree.created if channel and channel.ricecooker_version else \
+            descendants.filter(changed=True) \
+                .aggregate(latest_update=Max('modified')) \
+                .get('latest_update')
+
+        # See if the latest cached data is up to date since the last update to the channel
```

Comment:
Looks like we don't need to check here anymore? Can probably remove this comment